src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -6532,14 +6539,21 @@ void CodeGen::genJmpMethod(GenTree* jmp)
             regSet.AddMaskVars(genRegMask(argReg));
             gcInfo.gcMarkRegPtrVal(argReg, loadType);
 
-            if (compiler->lvaIsMultiregStruct(varDsc, compiler->info.compIsVarArgs))
+            if (varDsc->GetOtherArgReg() < REG_STK)
             {
                 // Restore the second register.
-                argRegNext = genRegArgNext(argReg);
+                argRegNext = varDsc->GetOtherArgReg();
+
+                if (emitter::isFloatReg(argRegNext))
```

Comment:
And why special case for float is need?

Reply:
Because it cannot get the exact type by using `varDsc->GetLayout()->GetGCPtrType(1)` (only returns TYP_I_IMPL for all primitive and non-gc value type) and `GetEmitter()->emitIns_R_S` needs the exact type for the arguments.

If you want to update calling convention like ARM64, we can do. However, I think it is better to investigate after RISC-V is stable.